Packs/UpdateEntriesBySearch/Scripts/MarkAsEvidenceBySearch/MarkAsEvidenceBySearch.py

```diff
@@ -0,0 +1,215 @@
+                    yield from iterate_value(v)
+
+            elif isinstance(value, dict):
+                for k, v in value.items():
+                    yield from iterate_value(v)
+            else:
+                yield value
+
+        matched = None
+        for node_path in self.__node_paths:
+            for val in iterate_value(demisto.get(entry, node_path)):
+                s = to_string(val)
+                if s is not None:
+                    if self.__exclude_pattern and self.__exclude_pattern.search(s):
+                        return None
+
+                    if not matched:
+                        matched = self.__include_pattern.search(s)
+        return matched
+
+
+class Entry:
+    def __init__(self, entry: Dict[str, Any], match: Optional[re.Match]):
+        self.entry = entry
+        self.match = match
+
+
+def iterate_entries(incident_id: Optional[str], query_filter: Dict[str, Any],
+                    entry_filter: Optional[EntryFilter] = None) -> Iterator[Entry]:
+    """
+    Iterate war room entries
+
+    :param incident_id: The incident ID to search entries from.
+    :param query_filter: Filters to search entries.
+    :param entry_filter: Filters to filter entries.
+    :return: An iterator to retrieve entries.
+    """
+    query_filter = dict(**query_filter)
+    first_id = 1
+    while True:
+        query_filter['firstId'] = str(first_id)
+
+        ents = demisto.executeCommand('getEntries', assign_params(
+            id=incident_id,
+            filter=query_filter
+        ))
+        if not ents:
+            break
+
+        if is_error(ents[0]):
+            if first_id == 1:
+                return_error('Unable to retrieve entries')
+            break
+
+        for ent in ents:
+            if not entry_filter:
+                yield Entry(ent, None)
+            else:
+                match = entry_filter.match(ent)
+                if match:
+                    yield Entry(ent, match)
+
+        # Set the next ID
+        last_id = ent['ID']
+        m = re.match('([0-9]+)', last_id)
+        if not m:
+            raise ValueError(f'Invalid entry ID: {last_id}')
+        next_id = int(m[1]) + 1
+        if next_id <= first_id:
+            break
+        first_id = next_id
+
+
+def main():
+    args = demisto.args()
+
+    build_pattern_args = assign_params(
+        pattern_algorithm=args.get('algorithm', 'raw'),
+        case_insensitive=args.get('case_insensitive', False)
+    )
+    include_pattern = build_pattern(**build_pattern_args, pattern=args['pattern'])
+    exclude_pattern = None
+    if args.get('exclude_pattern'):
+        exclude_pattern = build_pattern(**build_pattern_args, pattern=args['exclude_pattern'])
+
+    filter_options = argToList(args.get('filter_options', []))
+    output_option = args.get('summary', 'basic')
+
+    exclude_ids = []
+    if 'exclude_this_entry' not in filter_options:
+        exclude_ids.append(demisto.parentEntry()['id'])
```

Comment:
Can you please elaborate on the use in 'parentEntry'? What exactly are you expecting to get here?

Reply:
And are you sure regarding the logic here? if the 'exclude' is *not* in the filter you are adding it to the excluded ids?

Reply:
That's a bug. I've fixed it.

Reply:
I've used `demisto.parentEntry()['id']` to get the entry ID of itself.
For example, there is an entry which has 'AAA' in the warroom, and when you mark as note to it, then you run `MarkAsNoteBySearch pattern=AAA'.
In that case, two entries will be marked as note. First one is the entry you wanted to mark as note, and second one is the entry of `MarkAsNoteBySearch pattern=AAA' you've run.
In most of the case, you don't want to match to the second one. So it used `demisto.parentEntry()['id']` to identify the entry of itself.